README.md

```diff
@@ -126,6 +96,80 @@ This creates an EKS cluster with all possible configuration options defaulted.
 It is not intended for a production cluster.
 Refer to [Production cluster considerations](#production-cluster-considerations) for things to consider when creating a production cluster.
 
+
+### Migrating to current version of module from a version prior to 3.0.0
+
+TODO: Verify that I have covered all of https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1744#issuecomment-1027359982
+
+If you already have created an EKS cluster using a pre 3.0.0 version of this module there is unfortunately no easy 
+way to upgrade without recreating the cluster. If you already create the cluster in some other way and now set 
+`create_eks = false` you only n eed to remove some inputs. I won't cover that much simpler case here.
+
+While it would be a bit easier if you started using the same version of `terraform-aws-modules/eks/aws` as previously used by this module we 
+would advise against that. The reason is that this version is very old and doesn't support a lot of feature currently available with AWS.
+
+Let's say you created your 
+cluster using [an old version of the template](https://github.com/jx3-gitops-repositories/jx3-terraform-eks/tree/451bf5a1a453aca9a384a9f817d8b347e18c4c04) and change 
+your configuration to a [current version](https://github.com/jx3-gitops-repositories/jx3-terraform-eks). If you then run `terraform plan` you will see that basically 
+everything would be destroyed and then created. To mitigate that you can move resources in the terraform state to the new addresses. In some cases there are no 
+corresponding new address, instead you are better off removing resources to avoid that they get destroyed before the new resources are created. This means that you 
+need to remove those cloud resources manually later. You can also tweak configurations to prevent resources from be 
+replaced. If you check the output from `terraform plan` you will see that resources marked as "must be replaced" 
+have one or more inputs with the comment "# forces replacement". If it is a resource that you need to keep to 
+prevent disruption or data loss you should try to tweak the configuration so that the inputs value is reverted to 
+what it was before.
+
+```shell
+terraform state mv module.eks-jx.random_pet.current  random_pet.current # Only needed if cluster_name wasn't specified
+terraform state mv module.eks-jx.module.cluster.module.vpc module.vpc # Only needed if create_vpc wasn't false
+terraform state mv module.eks-jx.module.cluster.module.eks module.eks
+terraform state mv 'module.eks.aws_iam_role.cluster[0]' 'module.eks.aws_iam_role.this[0]'
```

Comment:
module.eks-jx.module.cluster.module.eks.aws_iam_role.cluster maybe?

Reply:
No, since in the line above module.eks-jx.module.cluster.module.eks was moved to module.eks